spec/response-interceptors.spec.js

```diff
@@ -64,6 +67,64 @@ test('you can pass in a custom interceptor', (t) => {
   });
 });
 
+test('the default error handler throws on respsonse error with a custom interceptor', (t) => {
+  t.plan(2);
+
+  const UserFactory = fetchFactory.create({
+    url: 'http://www.api.com/users',
+    method: 'GET',
+    interceptors: {
+      response: (data) => ({ name: 'bob' }),
+    },
+  }, {
+    findAll: {},
+  });
+
+  const stub = nock('http://www.api.com')
+    .get('/users')
+    .reply(404);
+
+  UserFactory.findAll().catch((error) => {
+    t.ok(stub.isDone(), 'the stub was called');
+    t.equal(error.response.status, 404);
+  });
+});
+
+test('the default error handler is not called if rejectOnBadResponse is set to false', (t) => {
```

Comment:
I think a clearer test for this would be to set `rejectOnBadResponse: false`, and assert that a 404 request resolves rather than rejects. That's easier to follow that using another custom interceptor I think.